configs/components/nssm.rb

```diff
@@ -1,14 +1,14 @@
 component "nssm" do |pkg, settings, platform|
-  pkg.version "2.24"
-  pkg.md5sum "b2edd0e4a7a7be9d157c0da0ef65b1bc"
-  pkg.url "https://nssm.cc/release/nssm-#{pkg.get_version}.zip"
+  pkg.load_from_json("configs/components/nssm.json")
 
-  # Because we're unpacking a zip archive, we need to set the path to the executable.
-  # We don't automatically have this set on windows, unfortunately. We need to set the
-  # path to have access to 7za.
-  pkg.environment "PATH" => "/cygdrive/c/ProgramData/chocolatey/tools:$$PATH" if platform.is_windows?
+  build_arch = platform.architecture == "x64" ? "x64" : "Win32"
 
-  win_arch = platform.architecture == "x64" ? "win64" : "win32"
+  pkg.install do
+    [
+      "/cygdrive/c/Program\\ files\\ \\(x86\\)/Microsoft\\ Visual\\ Studio/2017/BuildTools/Common7/Tools/vsdevcmd.bat",
+      "/cygdrive/c/Program\\ Files\\ \\(x86\\)/Microsoft\\ Visual\\ Studio/2017/BuildTools/MSBuild/15.0/Bin/msbuild.exe nssm.vcxproj /detailedsummary /p:Configuration=Release /p:OutDir=./out /p:Platform=#{build_arch} /p:PlatformToolset=v141 /p:TargetPlatformVersion=8.1",
```

Comment:
A few things here:

* `msbuild.exe` shouldn't be fully qualified
* Can we use the env vars for `%ProgramFiles(x86)%` location?  I'm actually surprised that works on the 32-bit host as `vsdevcmd` is under `%ProgramFiles%` there.
* Can we parameterize `PlatformToolset` and `TargetPlatformVersion` and move them up in the file by `build_arch`?
* Note that the the binary is auto-versioned with `version.cmd` in the nssm repo -
 so we can't use `/p:ApplicationVersion=" here and should instead rely on the `version.cmd` behavior I think (which uses `git describe` - https://github.com/puppetlabs/nssm/blob/master/version.cmd)